### PR TITLE
Add option to search the lastUpdated date in table

### PR DESCRIPTION
**Pull request by jvfdunkley**
### :pushpin: References

-   **Issue:** Closes [CU-byxk0d Last updated search](https://app.clickup.com/t/byxk0d)

### :memo: Implementation
- I added a DatePicker component from d2-ui-components. I added the lastUpdated property to the Filters interface in PredictorListPage as well as in GetPredictorsUseCase. The lastUpdated type had to be a Date object (or some kind of object, so I went with a Date) because the DatePicker component required it.
- I added lastUpdated to the list function in PredictorD2ApiRepository to search for greater than or equal to the lastUpdated date formatted in YYYY-MM-DD format.  

### :art: Screenshots
Screen Shot 2021-01-29 at 11 59 04 AM

### CODE REVIEW

**Comment by ifoche** [comment]
Task linked: [CU-byxk0d Last updated search](https://app.clickup.com/t/byxk0d)

File: src/webapp/pages/predictor-list/PredictorListPage.tsx [hunk 1]
```diff
@@ -219,8 +220,12 @@ export const PredictorListPage: React.FC = () => {
...
                 tableProps.onChangeSearch(update.search ?? "");
             }
-
-            setState(state => ({ ...state, ...update }));
+            if (update.lastUpdated) {
+                const convertedLastUpdated = new Date(update.lastUpdated);
+                setState(state => ({ ...state, lastUpdated: convertedLastUpdated }));
+            } else {
+                setState(state => ({ ...state, ...update }));
+            }
```
Comment (SferaDev) [inline src/webapp/pages/predictor-list/PredictorListPage.tsx]:
```suggestion

            setState(state => ({ ...state, ...update }));
```

File: src/webapp/pages/predictor-list/PredictorListPage.tsx [hunk 2]
```diff
@@ -251,6 +256,13 @@ export const PredictorListPage: React.FC = () => {
                             onChange={predictorGroups => onChangeFilter({ predictorGroups })}
                             label={i18n.t("Predictor groups")}
                         />
+
+                        <DatePicker
+                            placeholder={i18n.t("Last updated date")}
+                            value={state.lastUpdated ?? null}
+                            onChange={lastUpdated => onChangeFilter({ lastUpdated })}
+                            isFilter={true}
+                        />
```
Comment (SferaDev) [inline src/webapp/pages/predictor-list/PredictorListPage.tsx]:
```suggestion
                        <DatePicker
                            placeholder={i18n.t("Last updated date")}
                            value={state.lastUpdated ?? null}
                            isFilter={true}
                            onChange={(lastUpdated: { toDate(): Date }) =>
                                onChangeFilter({ lastUpdated: lastUpdated.toDate() })
                            }
                        />
```

**Comment by SferaDev** [comment]
Thanks @jvfdunkley, looks great!

Just a minor change to make it more functional, you can review it in the comments.

File: src/webapp/pages/predictor-list/PredictorListPage.tsx (hunk 2 above)
Comment (jvfdunkley) [reply src/webapp/pages/predictor-list/PredictorListPage.tsx]: Thanks that makes sense

**Pull request merged**